### PR TITLE
Added the can sender prototype

### PR DESCRIPTION
**Pull request by Zarar3**
For: https://github.com/waterloo-rocketry/2025-software-issues/issues/114

https://github.com/user-attachments/assets/fa4805f1-89f0-412d-a020-b1aa9e2ae346

image

## Summary by CodeRabbit

* **New Features**
  * Added a CAN message send form with validation and SEND action; introduced form primitives and new UI building blocks (button-group, separator).

* **Tests**
  * Added tests covering the messaging form UI, validation, placeholders, and submit payloads.

* **Style**
  * Formatting and quoting consistency improvements across UI components.

* **Chores**
  * Updated runtime dependencies and improved .gitignore rules for env/temp files.

[Review Change Stack](https://app.coderabbit.ai/change-stack/waterloo-rocketry/omnibus-avionics-testing-app/pull/7?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by john-jpet** [comment]
A few things to tackle in follow-up PRs:

- Wire up actual sending logic via IPC to the Omnibus backend (CAN/Commands channel)
- Replace free-text inputs with dropdowns using RocketCAN enums (e.g. msg_prio: HIGHEST/HIGH/MEDIUM/LOW, msg_type: -ACTUATOR_CMD/RESET_CMD/etc., board_type_id and board_inst_id from the Board IDs spec)
- Dynamic payload fields based on selected msg_type (e.g. ACTUATOR_CMD should expose actuator_id and actuator_state fields)
- Add send confirmation/error feedback in the UI once IPC is wired up

**Comment by ChrisYx511** [comment]
let's fix the merge conflicts!

**Comment by john-jpet** [comment]
> A few things to tackle in follow-up PRs:
> 
>     * Wire up actual sending logic via IPC to the Omnibus backend (CAN/Commands channel)
> 
>     * Replace free-text inputs with dropdowns using RocketCAN enums (e.g. msg_prio: HIGHEST/HIGH/MEDIUM/LOW, msg_type: -ACTUATOR_CMD/RESET_CMD/etc., board_type_id and board_inst_id from the Board IDs spec)
> 
>     * Dynamic payload fields based on selected msg_type (e.g. ACTUATOR_CMD should expose actuator_id and actuator_state fields)
> 
>     * Add send confirmation/error feedback in the UI once IPC is wired up

Bumping this

File: packages/renderer/tests/CanSender.test.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+        const timeInput = screen.getByLabelText(/time/i)
+        const submitButton = screen.getByRole('button', { name: /send/i })
+
+        fireEvent.change(timeInput, { target: { value: 'notanumber' } })
+        fireEvent.click(submitButton)
+
+        await waitFor(() => {
+            expect(screen.getByText(/Must be a valid integer/i)).toBeDefined()
+        })
```
[the remark this answers is not in the document]
Comment (Vaishali-Anapayan) [reply packages/renderer/tests/CanSender.test.tsx:45]: Not a big issue to fix in my opinion, unless you are really against these edge cases happening.

File: packages/renderer/tests/CanSender.test.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+        await waitFor(() => {
+            expect(consoleSpy).toHaveBeenCalledWith(expect.objectContaining({
+                msg_type: '{not valid json}',
+                msg_prio: 'anything goes',
+            }))
+        })
+
+        consoleSpy.mockRestore()
+    })
```
[the remark this answers is not in the document]
Comment (Vaishali-Anapayan) [reply packages/renderer/tests/CanSender.test.tsx:111]: CodeRabbit says it is only a minor issue and will only cause problems if a test assertion fails before the manual `mockRestore()` call. I'm still learning about mocking, so I don't know what the potential downsides are of making the suggested change. I'll leave the decision of whether to do it up to youu

**Review by Vaishali-Anapayan — APPROVED** [review]
Looks good to me! There are a couple of CodeRabbit nitpicks, but they are minor. Pulled the code and works great, ran the tests and they all passed.

Just as a note, when I pulled the code, I had to take out `"baseUrl": ".",` from tsconfig.app.json in packages/rednerer since I get this error: `Option 'baseUrl' is deprecated and will stop functioning in TypeScript 7.0. Specify compilerOption '"ignoreDeprecations": "6.0"' to silence this error.`. I have Typescript 5.9.3, so I'm not sure why I am getting the error on my end, but the code works fine when I take the line out. I'm pretty sure because 'paths' is specified. Anyway, nothing wrong with your code, just wanted to note that I did that on my end to run your code.
